weave/trace/vals.py

```diff
@@ -536,9 +552,22 @@ def make_trace_obj(
 
     if not isinstance(val, Traceable):
         if isinstance(val, ObjectRecord):
-            return WeaveObject(
-                val, ref=new_ref, server=server, root=root, parent=parent
-            )
+            res = WeaveObject(val, ref=new_ref, server=server, root=root, parent=parent)
```

Comment:
Why not initialize a Dataset here instead?

Reply:
good point -- we're special casing our known objects here so may as well just return the actual `Dataset`